mock_ros2_project/publisher_node.py

```diff
@@ -0,0 +1,71 @@
+#!/usr/bin/env python3
+"""
+ROS2 Publisher Node Example
+
+This node demonstrates how to create a simple publisher that sends
+string messages at regular intervals to demonstrate basic pub/sub communication.
+"""
+
+import rclpy
+from rclpy.node import Node
+from std_msgs.msg import String
+
+
+class PublisherNode(Node):
+    """Simple publisher node that publishes string messages.
+    
+    This node creates a publisher that sends string messages containing
+    a counter value at regular intervals.
+    """
+    
+    def __init__(self) -> None:
+        """Initialize the publisher node."""
+        super().__init__('publisher_node')
+        
+        # Create publisher for string messages
+        self.publisher_ = self.create_publisher(String, 'send_topic', 10)
+        
+        # Create timer to publish messages periodically
+        timer_period = 0.5  # seconds
+        self.timer = self.create_timer(timer_period, self.timer_callback)
```

Comment:
The publish period is fixed to 0.5 s. Declaring a parameter (e.g. `declare_parameter('timer_period', 0.5)`) and reading it here lets users adjust the rate at runtime or via launch files, improving flexibility.

```suggestion
        # Create timer to publish messages periodically
        timer_period = self.declare_parameter('timer_period', 0.5).get_parameter_value().double_value  # seconds
        self.timer = self.create_timer(timer_period, self.timer_callback)
```